File: JuegoAdivinarNúmeroPy/adivinarNumero/adivinarNumero.py
import random

numero_secreto = random.randint(1, 100)
print("¡Bienvenido al juego de Adivinar el Número!")
print("He pensado en un número entre 1 y 100. ¿Puedes adivinar cuál es?")
print("Tienes 10 intentos para adivinar el número. ¡Buena suerte!")

encontrado = False
intentos = 0
intentos_maximos = 10

while not encontrado and intentos < intentos_maximos:
    intento = int(input("Introduce tu número: "))
    intentos += 1

    diferencia = abs(numero_secreto - intento)

    if intento == numero_secreto:
        print(f"¡Felicidades! Has adivinado el número en {intentos} intentos.")
        encontrado = True
    elif diferencia > 20:
        print("Estás muy lejos.")
    elif diferencia > 10:
        print("Estás algo lejos.")
    elif diferencia > 5:
        print("Estás cerca.")
    else:
        print("Estás muy cerca.")

    if intento < numero_secreto:
        print("El número es más alto. Intenta de nuevo.")
    elif intento > numero_secreto:
        print("El número es más bajo. Intenta de nuevo.")

if not encontrado:
    print(f"Lo siento, has agotados tus {intentos_maximos} intentos. El número era {numero_secreto}.")

File: JuegoAdivinarNúmeroJava/adivinarNumero/src/adivinarnumero/adivinarNumero.java

package adivinarnumero;

import java.util.Scanner;
import java.util.Random;

public class adivinarNumero {
    public static void main(String[] args) {
        Scanner entrada = new Scanner(System.in);
        Random random = new Random();
        int numeroAleatorio = random.nextInt(100) + 1;
        
        boolean numeroAdivinado = false;
        int intentos = 0;
        int intentosMaximos = 10;
        
        System.out.println("¡Bienvenido al juego de adivinar el número!");
        System.out.println("He pensado en un número entre 1 y 100. ¿Puedes adivinar cuál es?");
        System.out.println("Tienes 10 intentos para adivinar el número. ¡Buena suerte!");
        System.out.println(numeroAleatorio);
        
        while (!numeroAdivinado || intentos < intentosMaximos) {
            System.out.println("Introduce tu número: ");
            int intento = entrada.nextInt();
            intentos += 1;
            
            if (intentos == intentosMaximos) {
            System.out.println("Lo siento, has agotados tus " + intentosMaximos + " intentos. El número era "+ numeroAleatorio);
            break;
            }
            
            if (intento < numeroAleatorio) {
                System.out.println("El número es más alto. Intenta de nuevo.");
            } else if (intento > numeroAleatorio) {
                System.out.println("El número es más bajo. Intenta de nuevo.");
            } else if (intento == numeroAleatorio) {
                System.out.println("¡Felicidades! Has adivinado el número en " + intentos + " intentos.");
                break;
            }
        }
        entrada.close();
    }
}
